File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatajournalentries.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDDataJournalEntriesPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDDataJournalEntriesQueryParams struct {
	OrderBy  *string `queryParam:"style=form,explode=true,name=orderBy"`
	Page     int     `queryParam:"style=form,explode=true,name=page"`
	PageSize *int    `queryParam:"style=form,explode=true,name=pageSize"`
	Query    *string `queryParam:"style=form,explode=true,name=query"`
}

type GetCompaniesCompanyIDDataJournalEntriesSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataJournalEntriesRequest struct {
	PathParams  GetCompaniesCompanyIDDataJournalEntriesPathParams
	QueryParams GetCompaniesCompanyIDDataJournalEntriesQueryParams
	Security    GetCompaniesCompanyIDDataJournalEntriesSecurity
}

type GetCompaniesCompanyIDDataJournalEntriesResponse struct {
	CodatDataContractsDatasetsJournalEntryPagedResponseModel *shared.CodatDataContractsDatasetsJournalEntryPagedResponseModel
	ContentType                                              string
	StatusCode                                               int
}

File: python-client-sdk/src/sdk/models/operations/get_companies_companyid_data_journalentries.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_datasets_journalentrypagedresponsemodel as shared_codat_datacontracts_datasets_journalentrypagedresponsemodel
from ..shared import security as shared_security
from typing import Optional


@dataclasses.dataclass
class GetCompaniesCompanyIDDataJournalEntriesPathParams:
    company_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'companyId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetCompaniesCompanyIDDataJournalEntriesQueryParams:
    page: int = dataclasses.field(metadata={'query_param': { 'field_name': 'page', 'style': 'form', 'explode': True }})
    order_by: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'orderBy', 'style': 'form', 'explode': True }})
    page_size: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'pageSize', 'style': 'form', 'explode': True }})
    query: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'query', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetCompaniesCompanyIDDataJournalEntriesSecurity:
    api_key_auth: Optional[shared_security.SchemeAPIKeyAuth] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    codat_login: Optional[shared_security.SchemeCodatLogin] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class GetCompaniesCompanyIDDataJournalEntriesRequest:
    path_params: GetCompaniesCompanyIDDataJournalEntriesPathParams = dataclasses.field()
    query_params: GetCompaniesCompanyIDDataJournalEntriesQueryParams = dataclasses.field()
    security: GetCompaniesCompanyIDDataJournalEntriesSecurity = dataclasses.field()
    

@dataclasses.dataclass
class GetCompaniesCompanyIDDataJournalEntriesResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    codat_data_contracts_datasets_journal_entry_paged_response_model: Optional[shared_codat_datacontracts_datasets_journalentrypagedresponsemodel.CodatDataContractsDatasetsJournalEntryPagedResponseModel] = dataclasses.field(default=None)
    